purchases/src/main/kotlin/com/revenuecat/purchases/utils/OfferingVideoPredownloader.kt

```diff
@@ -0,0 +1,40 @@
+@file:OptIn(InternalRevenueCatAPI::class)
+
+package com.revenuecat.purchases.utils
+
+import android.content.Context
+import com.revenuecat.purchases.InternalRevenueCatAPI
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.common.canUsePaywallUI
+import com.revenuecat.purchases.common.checkIfVideoComponentIsEnabled
+import com.revenuecat.purchases.paywalls.components.VideoComponent
+import com.revenuecat.purchases.storage.DefaultFileRepository
+import com.revenuecat.purchases.storage.FileRepository
+
+internal class OfferingVideoPredownloader(
+    context: Context,
+    canShowPaywalls: Boolean = canUsePaywallUI,
+    videoComponentIsEnabled: () -> Boolean = ::checkIfVideoComponentIsEnabled,
```

Comment:
Hmm I wonder if we should just access `BuildConfig.ENABLE_VIDEO_COMPONENT` directly... I guess the reason for doing it like this is to test it? We can mock static constants using `mockk` like `mockkStatic(BuildConfig::class)`, then `every { BuildConfig.ENABLE_VIDEO_COMPONENT } returns true`. Having said that, I don't think it's a blocker, since this would probably go away soon in any case.